go-controller/pkg/allocator/pod/macs.go

```diff
@@ -0,0 +1,157 @@
+	for owner, mac := range infraMACs {
+		if err := allocator.macManager.Reserve(owner, mac); err != nil {
+			return fmt.Errorf("failed to reserve infra MAC (%s) for owner (%s) on network (%s): %w",
+				mac, owner, networkName, err)
+		}
+		klog.V(5).Infof("Reserved MAC (%s) for infra (%s) on network (%s)", mac, owner, networkName)
+	}
+	for owner, mac := range podMACs {
+		if rerr := allocator.macManager.Reserve(owner, mac); rerr != nil {
+			return fmt.Errorf("failed to reserve pod MAC (%s) for owner (%s) on network (%s): %w",
+				mac, owner, networkName, rerr)
+		}
+		klog.V(5).Infof("Reserved MAC (%s) for pod (%s) on network (%s)", mac, owner, networkName)
+	}
+
+	return nil
+}
+
+// calculateSubnetsInfraMACAddresses return map of the network infrastructure mac addresses and owner name.
+// It calculates the gateway (.2) and management (.1) ports MAC addresses from their IP address.
+func calculateSubnetsInfraMACAddresses(subnets []config.CIDRNetworkEntry) map[string]net.HardwareAddr {
+	reservedMACs := map[string]net.HardwareAddr{}
+	for _, subnet := range subnets {
+		if subnet.CIDR == nil {
+			continue
+		}
+
+		gwIP := util.GetNodeGatewayIfAddr(subnet.CIDR)
+		gwMAC := util.IPAddrToHWAddr(gwIP.IP)
+		gwKey := fmt.Sprintf("gw-v%s", k8snet.IPFamilyOf(gwIP.IP))
+		reservedMACs[gwKey] = gwMAC
+
+		mgmtIP := util.GetNodeManagementIfAddr(subnet.CIDR)
+		mgmtMAC := util.IPAddrToHWAddr(mgmtIP.IP)
+		mgmtKey := fmt.Sprintf("mgmt-v%s", k8snet.IPFamilyOf(mgmtIP.IP))
+		reservedMACs[mgmtKey] = mgmtMAC
+	}
+
+	return reservedMACs
+}
+
+// fetchNetworkPods fetch pods in to the network NAD namespaces.
+func (allocator *PodAnnotationAllocator) fetchNetworkPods() ([]*corev1.Pod, error) {
+	var netPods []*corev1.Pod
+	for _, ns := range allocator.netInfo.GetNADNamespaces() {
+		pods, err := allocator.podLister.Pods(ns).List(labels.Everything())
+		if err != nil {
+			return nil, fmt.Errorf("failed to list pods for namespace %q: %v", ns, err)
+		}
+		for _, pod := range pods {
+			if pod == nil {
+				continue
+			}
+			if pod.Status.Phase != corev1.PodRunning || !pod.DeletionTimestamp.IsZero() && len(pod.Finalizers) == 0 {
+				// skip pods who are non-running or about to dispose
+				continue
+			}
+			netPods = append(netPods, pod)
+		}
+	}
+	return netPods, nil
+}
+
+// getPodMACs collected the given pods used MAC addresses from the pod-network annotation.
+func getPodMACs(pods []*corev1.Pod) (map[string]net.HardwareAddr, error) {
+	podMACs := map[string]net.HardwareAddr{}
+	for _, pod := range pods {
+		podNetworks, err := util.UnmarshalPodAnnotationAllNetworks(pod.Annotations)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal pod annotations %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		for _, network := range podNetworks {
+			if network.Role != types.NetworkRoleInfrastructure {
+				// primary UDN network role is infrastructure-lock on primary UDNs only.
```

Comment:
isn't this comment wrong ? the default network role is "infrastructure-lock" on primary UDNs only, right ?